src/main/java/fr/insee/pearljam/batch/service/PilotageLauncherService.java

```diff
@@ -480,6 +489,32 @@ private Map<String, SurveyUnitType> extractAndValidatePilotageFromSamplProcessin
 			logger.log(Level.INFO, "Campaign {} does not exist in Pilotage", campaignId);
 			throw new ValidateException("Campaign does not exist in Pilotage DB");
 		}
+
+		Set<String> meshuggahIds = sampleProcessing.getQuestionnaires().getQuestionnaire().stream()
+				.flatMap(q -> q.getInformationsGenerales().getMetadonneesCommunication().getCommunicationTemplate().stream())
+				.map(CommunicationTemplate::getMeshuggahId)
+				.collect(Collectors.toSet());
+
+		Map<String, CommunicationTemplateType> templatesByMeshuggahId =
+				communicationTemplateDaoImpl.findByMeshuggahIds(meshuggahIds).stream()
+						.collect(Collectors.toMap(CommunicationTemplateType::getMeshuggahId, Function.identity()));
+
+		sampleProcessing.getQuestionnaires().getQuestionnaire().stream()
+				.flatMap(q -> q.getInformationsGenerales().getMetadonneesCommunication().getCommunicationTemplate().stream())
+				.filter(ct -> {
+					CommunicationTemplateType dbTemplate = templatesByMeshuggahId.get(ct.getMeshuggahId());
+					return dbTemplate == null || !dbTemplate.getCampaignId().equals(campaignId);
+				})
+				.findAny()
+				.ifPresent(ct -> {
+          try {
+            throw new ValidateException("The communication template is not linked to the campaign.");
+          } catch (ValidateException e) {
+            throw new RuntimeException(e);
+          }
+        });
```

Comment:
suggestion : could compare meshuggahId from sample with meshuggahId in DB (from template having campaignId)
=> could prevent use of templatesByMeshuggahId.get(ct.getMeshuggahId()) for EACH questionnaire